packages/runtime/src/state/modules/embedded-components.ts

```diff
@@ -0,0 +1,33 @@
+import { type Element } from '@makeswift/controls'
+import { Action } from '../actions'
+
+export type EmbeddedComponent = {
+  documentKey: string
+  type: string
+  key: string
+  locale: string | null
+  rootElement: Element
+}
+
+export type State = Map<string, EmbeddedComponent>
+
+export function getInitialState({
+  embeddedComponents = new Map(),
+}: { embeddedComponents?: Map<string, EmbeddedComponent> } = {}): State {
+  return embeddedComponents
+}
+
+export function getEmbeddedComponents(state: State): Map<string, EmbeddedComponent> {
+  return state
+}
+
+export function getEmbeddedComponent(state: State, key: string): EmbeddedComponent | null {
```

Comment:
Could we have a more specific name for the `key` param here? I assume the `key` is the unique combination of `type`, `key`, `locale`, etc.

Reply:
Good question. I'm using the generated `uuid` as the key here. Note that this is different than the `encoded` base64 key that we're using on the builder<>orion.

What do you think we should use to make this more clear?

Reply:
Thank you. We severely need a naming convention. Please let me know if I'm not following that. In this case. We're registering the embedded based on the `documentKey`. So, in this case, does it make sense to use `documentKey` here?

Reply:
Updated to `documentKey`